hooks/useFetchProjects.ts

```diff
@@ -58,14 +59,45 @@ interface CachedProjects {
 const cacheKey = 'projectsData';
 const apiEndpoint = '/api/projects'; // The new API endpoint
 
+const extractTagsFromProjects = (projects: RepoItem[]) => {
+  return projects.reduce((acc: string[], project) => {
+    const tags = project.languages?.edges?.map(edge => edge.node.name);
+    tags.forEach(tag => {
+      if (!acc.includes(tag)) acc.push(tag);
+    });
+    return acc;
+  }, []);
```

Comment:
As I see it now this function is doing the following: 

Running `forEach` + condition inside every loop iteration + adding items that passed the condition to an array, then returning the array. 

Sounds like the `filter` method dont you agree?
 In my opinion, refactoring this code to use `filter` might make it a bit more readable and shorter.

Do you agree?

Reply:
Thats exactly the logic applied here but thanks to reduce its more compact and performant.